### PR TITLE
refactor(*): remove Row::cmp && remove dead code SortMergeJoin

### PR DESCRIPTION
**Pull request by TennyZhuang**
Signed-off-by: TennyZhuang <zty0826@gmail.com>

I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

Remove `Row::cmp`. `Row` can't be compared without order types. The `SortMergeJoin` implementation is not correct, `a ASC, b DESC, c ASC` is a possible case. It's also dead code now, remove it temporally.

## Checklist

- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Refer to a related PR or issue link (optional)

#7392

### CODE REVIEW

**Review by st1page — APPROVED** [review]
LGTM

File: src/common/src/row/mod.rs [hunk 1]
```diff
@@ -239,9 +215,9 @@ macro_rules! deref_forward_row {
             Row::eq(&(**this), other)
         }
 
-        fn cmp(this: &Self, other: impl Row) -> Ordering {
-            Row::cmp(&(**this), other)
-        }
+        // fn cmp(this: &Self, other: impl Row) -> Ordering {
```
Comment (BugenZhao) [inline src/common/src/row/mod.rs:218]: Also remove these commented lines?

**Review by BugenZhao — COMMENTED** [review]
Should we also remove the `impl Ord` for `OwnedRow`? I'm afraid there're a lot of references on that.

**Review by chenzl25 — APPROVED** [review]
LGTM

**Pull request merged**